packages/cosmwasm/src/multisigcosmwasmclient.spec.ts

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { makeCosmoshubPath, Secp256k1HdWallet } from "@cosmjs/launchpad";
+import { assert, sleep } from "@cosmjs/utils";
+
+import { MultisigCosmWasmClient, Vote } from "./multisigcosmwasmclient";
+import {
+  alice,
+  deployedCw3,
+  launchpad,
+  makeRandomAddress,
+  pendingWithoutCw3,
+  pendingWithoutLaunchpad,
+} from "./testutils.spec";
+
+describe("MultisigCosmWasmClient", () => {
+  describe("constructor", () => {
+    it("can be constructed", async () => {
+      const wallet = await Secp256k1HdWallet.fromMnemonic(alice.mnemonic);
+      const client = new MultisigCosmWasmClient(
+        launchpad.endpoint,
+        alice.address0,
+        wallet,
+        deployedCw3.instances[0],
+      );
+      expect(client).toBeTruthy();
+    });
+  });
+
+  describe("queries", () => {
+    it("works", async () => {
+      pendingWithoutLaunchpad();
+      pendingWithoutCw3();
+      const wallet = await Secp256k1HdWallet.fromMnemonic(alice.mnemonic);
+      const client = new MultisigCosmWasmClient(
+        launchpad.endpoint,
+        alice.address0,
+        wallet,
+        deployedCw3.instances[0],
+      );
+      const result = await client.queryContractSmart(deployedCw3.instances[0], { threshold: {} });
```

Comment:
I think it would be nice to expose query functions from `MultisigCosmWasmClient`, which allow doing the most important CW3 queries directly. This would internalize the query message format `{ threshold: {} }`.